src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AuthorizationApiServiceImpl.scala

```diff
@@ -107,11 +108,12 @@ final case class AuthorizationApiServiceImpl(
 
   private def upsertTenantBySelfcareId(selfcareId: String)(implicit contexts: Seq[(String, String)]): Future[Tenant] =
     for {
-      partyInstitution <- partyProcess.getInstitution(selfcareId)
-      _                <- assertTenantAllowed(selfcareId, partyInstitution.origin)
-      tenant           <- tenantProcessService
-        .selfcareUpsertTenant(partyInstitution.origin, partyInstitution.originId, partyInstitution.description)(
-          partyInstitution.id.toString
+      selfcareUuid <- selfcareId.toFutureUUID
+      institution  <- selfcareV2ClientService.getInstitution(selfcareUuid).map(_.toApi).flatMap(_.toFuture)
```

Comment:
Is this conversion useful?
```suggestion
      institution  <- selfcareV2ClientService.getInstitution(selfcareUuid)
```

Reply:
Yes because selfcare yields Option value